src/AbstractNavItem.js

```diff
@@ -47,8 +50,9 @@ const AbstractNavItem = React.forwardRef(
       if (!props.role && navContext.role === 'tablist') props.role = 'tab';
 
       props['data-rb-event-key'] = navKey;
-      props.id = navContext.getControllerId(navKey);
-      props['aria-controls'] = navContext.getControlledId(navKey);
+      props.id = navContext.getControllerId(navKey) || props.id;
```

Comment:
shouldn't the explicitly-specified values take precedence over calculated ones?

Reply:
hmm normally yes, but in this case, if there is a context provided id the user would have to provide the id in both places otherwise you get a mismatch and it's broken. We used to warn saying the ID is ignored, and that one should use generateId or whatever it is on the parent to control ids

Reply:
what should we do about https://github.com/react-bootstrap/react-bootstrap/issues/3940, then?

Reply:
maybe just warn again?

Reply:
hmm, would this actually fix that issue? because the navcontext won't assign a controller ID?

Reply:
we should always use the controller id, might just want to surface to a user that we are ignoring their props id?